### PR TITLE
Issue #185 is fixed. 

### LINKED ISSUE #185: Save temporary polygons in remove artefacts
When the user uses the tool _remove artefacts_, it would be good to have an option to save the polygons he draws. We could implement that at the end, when the user clicks _Run_, a warning pops up reminding him that the polygons will be lost and giving the option to make them permanent and specify an output file path.

### PR DESCRIPTION
**Pull request by jaminzoda**
The polygons drawn with the Remove artefacts tool can now be saved into a shapefile.
Close #185

### CODE REVIEW

File: gui/remove_arts_dlg.py [hunk 1]
```diff
@@ -37,6 +39,26 @@ def defineParameters(self):
...
+                                                              label = "Save mask layer.")
+
+        self.masksOutputPath = self.addAdvancedParameter(QgsFileWidget,
+                                                    label = "Output file path:")
+        self.masksOutputPath.setStorageMode(self.masksOutputPath.SaveFile)
+        self.masksOutputPath.setFilter('*.shp')
+        default_file_path = os.path.join(tempfile.gettempdir(),"remove_artefacts_polygons.shp")
+        if len(default_file_path)>68:
+            d_path, f_path = os.path.split(default_file_path)
+            while True:
```
Comment (jaminzoda) [inline gui/remove_arts_dlg.py:52]: Replace with a for loop to avoid getting into an infinite loop.

**Review by jaminzoda — COMMENTED** [review]
Default file path for saving drawn polygons must use for loop instead of a while loop, which may cause an infinite loop.

**Comment by Diego635** [comment]
Couldn't load plugin 'terra_antiqua' due to an error when calling its classFactory() method 

TypeError: tuple expected at most 1 arguments, got 2 
Traceback (most recent call last):
  File "/Applications/QGIS3.10.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 334, in _startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/terra_antiqua/__init__.py", line 35, in classFactory
    from .core.terra_antiqua import TerraAntiqua
  File "/Applications/QGIS3.10.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 792, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/terra_antiqua/core/terra_antiqua.py", line 44, in 
    from .compile_tb import TaCompileTopoBathy
  File "/Applications/QGIS3.10.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 792, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/terra_antiqua/core/compile_tb.py", line 19, in 
    from .utils import (
  File "/Applications/QGIS3.10.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 792, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/terra_antiqua/core/utils.py", line 1224, in 
    class TaVectorFileWriter(QgsVectorFileWriter):
  File "/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/terra_antiqua/core/utils.py", line 1231, in TaVectorFileWriter
    driverName:str) -> tuple(QgsVectorFileWriter.WriterError, str):
TypeError: tuple expected at most 1 arguments, got 2


Python version: 3.7.3 (default, Mar 27 2019, 09:23:15) [Clang 10.0.1 (clang-1001.0.46.3)] 
QGIS version: 3.10.14-A Coruña A Coruña, 8374282d2a 

Python Path:
/Applications/QGIS3.10.app/Contents/MacOS/../Resources/python
/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python
/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
/Applications/QGIS3.10.app/Contents/MacOS/../Resources/python/plugins
/Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7/lib-dynload
/Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7
/Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7/site-packages/geos
/Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python3.7/site-packages
/Applications/QGIS3.10.app/Contents/Resources/python
/Applications/QGIS3.10.app/Contents/Frameworks/Python.framework/Versions/Current/lib/python37.zip
/Users/diego/Library/Application Support/QGIS/QGIS3/profiles/default/python
/Users/diego/Downloads/TA_sample_data_bundle

**Comment by Diego635** [comment]
works.
Ready to be merged.
The only thing to discuss is that the shapefile created with the polygons from remove artefacts tool is not loaded into the canvas. Maybe it should?
This does not affect the functionality of the tool so it can be merged unto the release candidate

**Pull request merged**